src/librustc_lint/builtin.rs

```diff
@@ -1915,38 +1918,47 @@ impl Async2018 {
             }
         }
     }
-    fn report(&mut self, cx: &EarlyContext, span: Span) {
-        // don't lint `r#async`
-        if cx.sess.parse_sess.raw_identifier_spans.borrow().contains(&span) {
-            return;
-        }
-        let mut lint = cx.struct_span_lint(
-            ASYNC_IDENTS,
-            span,
-            "`async` is a keyword in the 2018 edition",
-        );
-
-        // Don't suggest about raw identifiers if the feature isn't active
-        lint.span_suggestion_with_applicability(
-            span,
-            "you can use a raw identifier to stay compatible",
-            "r#async".to_string(),
-            Applicability::MachineApplicable,
-        );
-        lint.emit()
-    }
 }
 
-impl EarlyLintPass for Async2018 {
+impl EarlyLintPass for KeywordIdents {
     fn check_mac_def(&mut self, cx: &EarlyContext, mac_def: &ast::MacroDef, _id: ast::NodeId) {
         self.check_tokens(cx, mac_def.stream());
     }
     fn check_mac(&mut self, cx: &EarlyContext, mac: &ast::Mac) {
         self.check_tokens(cx, mac.node.tts.clone().into());
     }
     fn check_ident(&mut self, cx: &EarlyContext, ident: ast::Ident) {
-        if ident.as_str() == "async" {
-            self.report(cx, ident.span);
+        let next_edition = match cx.sess.edition() {
+            Edition::Edition2015 => {
+                match &ident.as_str()[..] {
+                    "async" |
+                    "try" => 2018,
```

Comment:
Feels like this should be using `Edition::Edition2018`, which already implements `Display`.